core/vec3.cpp

```diff
@@ -1,218 +1,133 @@
-
-    Vec3 operator*(float num, const Vec3 &vec)
-    {
-        return Vec3(vec.x() * num, vec.y() * num, vec.z() * num);
-    }
-
-    Vec3 operator*(const Vec3 &vec, float num)
-    {
-        return Vec3(vec.x() * num, vec.y() * num, vec.z() * num);
-    }
-
-#pragma endregion
-#pragma region OPERATOR_MULTIPLY_EQUAL
-
-    Vec3 &Vec3::operator*=(float num)
-    {
-        x_ *= num;
-        y_ *= num;
-        z_ *= num;
-        return *this;
-    }
-
-#pragma endregion
-#pragma region OPERATOR_DIVISION
-
-    Vec3 operator/(const Vec3 &vec, float num)
-    {
-        if (num <= constants::kAccuracy)
-        {
-            std::stringstream ss;
-            ss << "Divider of the Vec3 object can't be close or equal to 0. Object: " << vec << ", divider: " << num;
-            throw std::logic_error(ss.str().c_str());
-        }
-        else
-        {
-            return Vec3(vec.x() / num, vec.y() / num, vec.z() / num);
-        }
-    }
-
-#pragma endregion
-#pragma region METHODS
-
-    float Vec3::scalarProduct(const Vec3 &other) const
-    {
-        return x_ * other.x() + y_ * other.y() + z_ * other.z();
-    }
-    Vec3 Vec3::crossProduct(const Vec3 &other) const
-    {
-        return Vec3(y_ * other.z() - z_ * other.y(), z_ * other.x() - x_ * other.z(), x_ * other.y() - y_ * other.x());
-    }
-    float Vec3::magnitudeSquared() const
-    {
-        return x_ * x_ + y_ * y_ + z_ * z_;
-    }
-    float Vec3::magnitude() const
-    {
-        return std::sqrt(this->magnitudeSquared());
-    }
-    Vec3 Vec3::normalize() const
-    {
-        return (*this) / magnitude();
-    }
-
-#pragma endregion
-#pragma region SINGLE_OPERATORS
-
-    bool operator==(const Vec3 &left, const Vec3 &right)
-    {
-        return (std::abs(left.x() - right.x()) < constants::kAccuracy &&
-                std::abs(left.y() - right.y()) < constants::kAccuracy &&
-                std::abs(left.z() - right.z()) < constants::kAccuracy);
-    }
-
-    bool operator!=(const Vec3 &left, const Vec3 &right)
-    {
-        return !(left == right);
-    }
-    std::ostream &operator<<(std::ostream &os, const Vec3 &srcVec3)
-    {
-        return os << "Vec3(" << srcVec3.x() << ", " << srcVec3.y() << ", " << srcVec3.z() << ")";
-    }
-
-    Vec3 &Vec3::operator=(const Vec3 &other)
-    {
-        x_ = other.x();
-        y_ = other.y();
-        z_ = other.z();
-        return *this;
-    }
-
-#pragma endregion
-#pragma region GETTERS_SETTERS
-
-    void Vec3::setX(float num)
-    {
-        x_ = num;
-    }
-    float Vec3::x() const
-    {
-        return x_;
-    }
-    void Vec3::setY(float num)
-    {
-        y_ = num;
-    }
-    float Vec3::y() const
-    {
-        return y_;
-    }
-    void Vec3::setZ(float num)
-    {
-        z_ = num;
-    }
-    float Vec3::z() const
-    {
-        return z_;
-    }
-
-#pragma endregion
+#include <sstream>
+
+namespace core {
+
+Vec3::Vec3(std::initializer_list<float> initList) {
```

Comment:
I mentioned earlier somewhere, that this constructor is useless since you already can construct like this:

```
Vec3{1, 2, 3}
```

Reply:
yeah i remember about it, but deleting this will cause a lot of trouble, so thats why I dont want to do it for now. added to my TODO's :)

Reply:
What is "a lot of trouble"?

Reply:
I mean amout of erros is biger then I can possibly show on the picture. To change it it would take me more then 2h to do so, baecause there are so many objects in tests that are usign straightforward initializer_list

Reply:
It would take me 5 min to fix this across entire project, assuming the usage is `Vec3({1, 2, 3})` and you need to convert this to `Vec3{1, 2, 3}`. I think you just need to read how to use search and replace in VSCode. These is basic tool.

Reply:
This is not the only case. There are some cases with only list or some implementation requeirs me to add core::Vec3. I know that core::Vec3({1,2,3}) is very bad, I will fix  that later, especially thatthis case ony happens in test, which will be removed anyway in the future, becuase they are badly implemented ;)